misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/TestAnnotationProcessor.java

```diff
@@ -137,7 +173,45 @@ public void process(FunctionNode functionNode, List<AnnotationAttachmentNode> an
                             vals[1] = value;
                         }
                     });
-                    suite.addMockFunction(vals[0] + MOCK_ANNOTATION_DELIMITER + vals[1], functionName);
+
+                    if (bLangPackage == null) {
+                        setBlangPackage(attachmentNode);
+                    }
+
+                    if (vals[0].isEmpty()) {
+                        diagnosticLog.logDiagnostic(Diagnostic.Kind.ERROR, ((BLangFunction) functionNode).pos,
+                                "Module name cannot be empty");
+                    }
+
+                    if (vals[1].isEmpty()) {
+                        diagnosticLog.logDiagnostic(Diagnostic.Kind.ERROR, ((BLangFunction) functionNode).pos,
+                                "Function name cannot be empty");
+                    }
+
+                    SymbolTable symbolTable = SymbolTable.getInstance(compilerContext);
+                    Map<BPackageSymbol, SymbolEnv> packageEnvironmentMap = symbolTable.pkgEnvMap;
```

Comment:
You don't need to do this each time the process() method gets invoked right? I think this can just be done in the init.

Reply:
Moved to the init function